src/components/Evolution.ts

```diff
@@ -3,11 +3,61 @@ import { Component } from "eightbittr/lib/Component";
 import { FullScreenPokemon } from "../FullScreenPokemon";
 import { IPokemon } from "./Battles";
 import { 
-    IPokemonEvolution, IPokemonEvolutionByLevel, IPokemonEvolutionByStats, 
-    IPokemonEvolutionRequirements
+    IPokemonEvolution, IPokemonEvolutionByItem, IPokemonEvolutionByLevel, 
+    IPokemonEvolutionByStats, IPokemonEvolutionByTrade, IPokemonEvolutionRequirement
 } from "./constants/Pokemon";
 
 /**
+ * Holds arguments used in IRequirementHander functions.
+ */
+export interface IHandlerArgs<T1, T2> {
```

Comment:
Template names like `T1` and `T2` aren't very descriptive. Choose a name that describes what it does (and add `@type` docs).